tests/modules/collaborations/test_models.py

```diff
@@ -21,116 +21,135 @@ def test_collaboration_create_with_members(
 
-    def set_read_approval_state(*user_guid_states):
-        for user_guid, state in user_guid_states:
-            collab.set_read_approval_state_for_user(user_guid, state)
-        for association in collab.collaboration_user_associations:
+        def set_read_approval_state(*user_guid_states):
             for user_guid, state in user_guid_states:
-                if association.user_guid == user_guid:
-                    assert association.read_approval_state == state
+                collab.set_read_approval_state_for_user(user_guid, state)
+            for association in collab.collaboration_user_associations:
+                for user_guid, state in user_guid_states:
+                    if association.user_guid == user_guid:
+                        assert association.read_approval_state == state
+
+        set_read_approval_state(
+            (collab_user_a.guid, CollaborationUserState.APPROVED),
+            (collab_user_b.guid, CollaborationUserState.DECLINED),
+        )
 
-    set_read_approval_state(
-        (collab_user_a.guid, CollaborationUserState.APPROVED),
-        (collab_user_b.guid, CollaborationUserState.DECLINED),
-    )
+        assert collab.get_read_state() == CollaborationUserState.DECLINED
 
-    assert collab.get_read_state() == CollaborationUserState.DECLINED
+        set_read_approval_state(
+            (collab_user_a.guid, CollaborationUserState.PENDING),
+            (collab_user_b.guid, CollaborationUserState.DECLINED),
+        )
 
-    set_read_approval_state(
-        (collab_user_a.guid, CollaborationUserState.PENDING),
-        (collab_user_b.guid, CollaborationUserState.DECLINED),
-    )
+        assert collab.get_read_state() == CollaborationUserState.DECLINED
 
-    assert collab.get_read_state() == CollaborationUserState.DECLINED
+        set_read_approval_state(
+            (collab_user_a.guid, CollaborationUserState.DECLINED),
+            (collab_user_b.guid, CollaborationUserState.PENDING),
+        )
 
-    set_read_approval_state(
-        (collab_user_a.guid, CollaborationUserState.DECLINED),
-        (collab_user_b.guid, CollaborationUserState.PENDING),
-    )
+        assert collab.get_read_state() == CollaborationUserState.DECLINED
 
-    assert collab.get_read_state() == CollaborationUserState.DECLINED
+        set_read_approval_state(
+            (collab_user_a.guid, CollaborationUserState.APPROVED),
+            (collab_user_b.guid, CollaborationUserState.APPROVED),
+        )
 
-    set_read_approval_state(
-        (collab_user_a.guid, CollaborationUserState.APPROVED),
-        (collab_user_b.guid, CollaborationUserState.APPROVED),
-    )
+        assert collab.get_read_state() == CollaborationUserState.APPROVED
 
-    assert collab.get_read_state() == CollaborationUserState.APPROVED
+    finally:
+        if collab:
+            collab.delete()
 
 
 def test_collaboration_edit_state_changes(db, collab_user_a, collab_user_b):
+    collab = None
+    try:
```

Comment:
These try blocks are just to ensure deletion of the collaboration in the event of errors, yes?